app/models/bishop.rb

```diff
@@ -1,18 +1,7 @@
 class Bishop < Piece
-  def valid_move?(destination_x, destination_y)
```

Comment:
I simplified the valid_move? function for almost all the pieces.  I found the repeated code dizzying and I tried to make it easy to read.